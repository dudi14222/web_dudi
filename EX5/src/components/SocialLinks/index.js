import React from 'react';
import './socialLinks.css';
const SocialLinks = () => (
    <div className="social-media">
        <ul className="list-inline">
            <li><a target="_blank" href="https://www.facebook.com/dudi.hamdi" className="icoFacebook" title="Facebook">
                <i className="fa fa-facebook"></i></a>
            </li>
            <li><a target="_blank" href="https://twitter.com/?lang=en" className="icoTwitter" title="Twitter">
                <i className="fa fa-twitter"></i></a>
            </li>
            <li><a target="_blank" href="https://plus.google.com/discover?hl=iw" className="icoGoogle" title="Google +">
                <i className="fa fa-google-plus"></i></a>
            </li>
            <li><a target="_blank" href="https://www.linkedin.com/" className="icoLinkedin" title="Linkedin">
                <i className="fa fa-linkedin"></i></a>
            </li>
        </ul>
    </div>
);
export default SocialLinks;
