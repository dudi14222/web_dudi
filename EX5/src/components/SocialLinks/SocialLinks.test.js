import SocialLinks from '../SocialLinks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('SocialLinks shallow tests', () => {

    it(`SocialLinks has facebook link`, () => {
        const socialLinks = shallow(<SocialLinks />)
        const fbLink = socialLinks.find('.icoFacebook'); 
        expect(fbLink.prop("href")).toBe("https://www.facebook.com/dudi.hamdi");    
    })

    it(`SocialLinks has Linkedin link`, () => {
        const socialLinks = shallow(<SocialLinks />)
        const linkedin = socialLinks.find('.icoLinkedin'); 
        expect(linkedin.prop("href")).toBe("https://www.linkedin.com/");    
    })

})

describe('SocialLinks snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <SocialLinks />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

