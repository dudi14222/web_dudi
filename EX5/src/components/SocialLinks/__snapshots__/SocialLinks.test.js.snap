// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks snapshot tests test snapshot 1`] = `
<div
  className="social-media"
>
  <ul
    className="list-inline"
  >
    <li>
      <a
        className="icoFacebook"
        href="https://www.facebook.com/dudi.hamdi"
        target="_blank"
        title="Facebook"
      >
        <i
          className="fa fa-facebook"
        />
      </a>
    </li>
    <li>
      <a
        className="icoTwitter"
        href="https://twitter.com/?lang=en"
        target="_blank"
        title="Twitter"
      >
        <i
          className="fa fa-twitter"
        />
      </a>
    </li>
    <li>
      <a
        className="icoGoogle"
        href="https://plus.google.com/discover?hl=iw"
        target="_blank"
        title="Google +"
      >
        <i
          className="fa fa-google-plus"
        />
      </a>
    </li>
    <li>
      <a
        className="icoLinkedin"
        href="https://www.linkedin.com/"
        target="_blank"
        title="Linkedin"
      >
        <i
          className="fa fa-linkedin"
        />
      </a>
    </li>
  </ul>
</div>
`;
